sympy/integrals/transforms.py

```diff
@@ -1008,9 +1008,33 @@ def repl(ex, *args):
 def _laplace_transform(f, t, s_, simplify=True):
     """ The backend function for Laplace transforms. """
     from sympy import (re, Max, exp, pi, Min, periodic_argument as arg_,
-                       arg, cos, Wild, symbols, polar_lift)
+                       arg, cos, Wild, symbols, polar_lift, limit,
+                       Add, DiracDelta)
     s = Dummy('s')
-    F = integrate(exp(-s*t) * f, (t, 0, oo))
+    zero_minus = symbols('zero_minus', real=True)
+    # Expressions with DiracDelta need to be computed with a directional
+    # limit, but this will fail on some sums, so we first use the
+    # linearity of the Laplace transform to split the integral into a sum
+    # of integrals.
+    if f.has(DiracDelta(t)):
```

Comment:
Will this work for something like `DiracDelta(2*t)`?

Reply:
> Will this work for something like `DiracDelta(2*t)`?

No :(

But fortunately the new part I wrote now makes it unnecessary to restrict the algorithm only to Dirac impulses at 0, so the solution is simply to use if `f.has(DiracDelta):`

Reply:
I have pushed a solution; github's check fail due to an internal misconfiguration, though.  Will you look at it again before the checks run, or shall I try and force the checks?` before the review?